### PR TITLE
Use Popovers and Submenus for Header Bar Buttons

### LINKED ISSUE #114: Use popover instead of popup for HeaderBar buttons
## Expected Behavior
The `Gtk.Button` widgets `Menu` and `Insert` should trigger a `Gtk.Popover` and show attached accelerators.
Example from Sequeler:
screenshot from 2019-03-05 20-09-26

## Current Behavior
They currently open a `Gtk.Popup` which is pretty ugly and doesn't allow listing Accelerators.

### PR DESCRIPTION
**Pull request by Alecaddd**
## Summary
Convert the currently implement `Gtk.Popup`s into `Gtk.Popover`s.
This is helpful and necessary since those dropdowns reveal multiple submenus, and navigating through them with a regular popup is prone to mistakes.
Popovers also allow us to better handle spacing and show accelerators.

## Screenshot
Screenshot from 2019-04-28 22-44-50

## Known Issues / Things To Do
- [x] Convert Main Menu
- [x] Convert Shapes Menu
- [x] Implement Submenu
- [x] Add Icons to menu items Submenu

- Fixes #114

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
What's the issue you're having with submenus?

File: src/Layouts/HeaderBar.vala [hunk 1]
```diff
@@ -62,35 +63,89 @@ public class Akira.Layouts.HeaderBar : Gtk.HeaderBar {
...
+        menu_popover_grid.margin_bottom = 3;
+        menu_popover_grid.orientation = Gtk.Orientation.VERTICAL;
+        menu_popover_grid.width_request = 220;
 
-        var new_window = new Gtk.MenuItem.with_label (_("New Window"));
-        new_window.action_name = Akira.Services.ActionManager.ACTION_PREFIX + Akira.Services.ActionManager.ACTION_NEW_WINDOW;
-        menu_items.add (new_window);
-        menu_items.add (new Gtk.SeparatorMenuItem ());
+        var new_window_button = new Akira.Partials.PopoverButton (
+            _("Open New Window"), {"<Ctrl>n"});
```
Comment (bilelmoussaoui) [inline src/Layouts/HeaderBar.vala]: Is there a reason for putting those on a new line?

Comment (Alecaddd) [reply src/Layouts/HeaderBar.vala]: I started formatting with a soft 80 character limits and a hard 100.
It looks a bit weird and I'll probably avoid it, tho.
What do you think?

**Comment by Alecaddd** [comment]
@bilelmoussaoui 
> What's the issue you're having with submenus?

I literally can't make it work. No idea why but when I click on the `Open recent` menu item, I get a warning saying that the element with the name I specified is not available in the stack.
Feel free to give it a look and see if I missed something or I'm using something incorrectly.

**Comment by bilelmoussaoui** [comment]
Let me take a look at that tomorrow.

**Comment by Alecaddd** [comment]
I was finally able to figure out how to implement submenus, yay!
Time to add some icons to make the menus look pretty, and then this PR will be ready for a full review.

**Comment by Alecaddd** [comment]
This is ready to be reviewed!
popovers

File: src/Layouts/HeaderBar.vala [hunk 2]
```diff
@@ -164,17 +118,181 @@ public class Akira.Layouts.HeaderBar : Gtk.HeaderBar {
...
+        grid.add (vector);
+        grid.add (pencil);
+        grid.add (text);
+        grid.add (image);
+        grid.show_all ();
+
+        var popover = new Gtk.PopoverMenu ();
+        popover.add (grid);
+        popover.add (shapes_grid);
+        popover.child_set_property (grid, "submenu", "main");
```
Comment (bilelmoussaoui) [inline src/Layouts/HeaderBar.vala]: Can we rename main to main-menu here?

Comment (Alecaddd) [reply src/Layouts/HeaderBar.vala]: I did that at the beginning but I'm getting a Gtk Warning saying the "main" submenu can't be found in the Gtk.Stack.
I wonder if PopoverMenu expects the main menu to be always called "main"

Comment (bilelmoussaoui) [reply src/Layouts/HeaderBar.vala]: Can you check using Inspector the name of each child, maybe something is missing somewhere?

Comment (Alecaddd) [reply src/Layouts/HeaderBar.vala]: Screenshot from 2019-05-06 21-49-51

Comment (Alecaddd) [reply src/Layouts/HeaderBar.vala]: I don't think this is a big deal as everything works as expected and the menu name is not visible to the user but only used by GTK. What do you think?

Comment (bilelmoussaoui) [reply src/Layouts/HeaderBar.vala]: Yeah it was just a nitpick. Let's merge this too.

**Pull request merged**